doc/en/explanation/goodpractices.rst

```diff
@@ -112,43 +96,24 @@ This has the following benefits:
     See :ref:`pytest vs python -m pytest` for more information about the difference between calling ``pytest`` and
     ``python -m pytest``.
 
-Note that this scheme has a drawback if you are using ``prepend`` :ref:`import mode <import-modes>`
-(which is the default): your test files must have **unique names**, because
-``pytest`` will import them as *top-level* modules since there are no packages
-to derive a full package name from. In other words, the test files in the example above will
-be imported as ``test_app`` and ``test_view`` top-level modules by adding ``tests/`` to
-``sys.path``.
-
-If you need to have test modules with the same name, you might add ``__init__.py`` files to your
-``tests`` folder and subfolders, changing them to packages:
+For the most convenient experience,
+choose the ``import-lib`` :ref:`import mode <import-modes>`.
+To this end, add the following to your ``pyproject.toml``:
 
-.. code-block:: text
-
-    pyproject.toml
-    setup.cfg
-    mypkg/
-        ...
-    tests/
-        __init__.py
-        foo/
-            __init__.py
-            test_view.py
-        bar/
-            __init__.py
-            test_view.py
+.. code-block:: toml
 
-Now pytest will load the modules as ``tests.foo.test_view`` and ``tests.bar.test_view``, allowing
-you to have modules with the same name. But now this introduces a subtle problem: in order to load
-the test modules from the ``tests`` directory, pytest prepends the root of the repository to
-``sys.path``, which adds the side-effect that now ``mypkg`` is also importable.
+    [tool.pytest.ini_options]
+    addopts = [
+        "--import-mode=importlib",
+    ]
 
-This is problematic if you are using a tool like `tox`_ to test your package in a virtual environment,
-because you want to test the *installed* version of your package, not the local code from the repository.
+The default :ref:`import mode <import-modes>` ``prepend`` has several drawbacks [1]_.
```

Comment:
Hmm, my understanding is that there‘s only the property that you can’t import from test files, which is an advantage in my book. It’s only a blocker for adoption by existing projects because it’s not backwards compatible, which I should mention.

Are there any disadvantages apart from “isn’t backwards compatible in this regard”?

Reply:
@flying-sheep gentle ping here just in case you missed this comment. 👍

Reply:
Sure, I’ll spell out the differences there! Due to the amount of misunderstandings of the good practices document in the past, I think there should be several references, so I think referring to that footnote/section here in some form isn’t the worst idea.